packages/terra-application-layout/src/header/_ApplicationHeader.jsx

```diff
@@ -0,0 +1,253 @@
+    }
+    return undefined;
+  }
+
+  handleUtilityDiscloseRequest(utility) {
+    this.setState({
+      utilityComponent: React.cloneElement(utility, { onRequestClose: this.handleUtilityPopupCloseRequest }),
+    });
+  }
+
+  handleUtilityPopupCloseRequest() {
+    if (this.state.utilityComponent) {
+      this.setState({ utilityComponent: undefined });
+    }
+  }
+
+  handleUtilityOnChange(event, key) {
+    const { utilityConfig, app } = this.props;
+
+    utilityConfig.onChange(event, key, app && app.disclose);
+  }
+
+  renderToggle() {
+    const { layoutConfig, intl } = this.props;
+
+    if (layoutConfig.toggleMenu) {
+      return (
+        <div className={cx('toolbar-toggle')}>
+          <button
+            className={cx('toggle-button')}
+            aria-label={intl.formatMessage({ id: 'Terra.applicationLayout.applicationHeader.menuToggleLabel' })}
+            onClick={layoutConfig.toggleMenu}
+          >
+            <IconMenu />
+          </button>
+        </div>
+      );
+    }
+
+    return null;
+  }
+
+  renderAppName() {
+    const { nameConfig } = this.props;
+
+    if (nameConfig.accessory || nameConfig.title) {
+      return (
+        <ApplicationHeaderName accessory={nameConfig.accessory} title={nameConfig.title} />
+      );
+    }
+
+    return null;
+  }
+
+  renderNavigation(isCompact) {
+    const { applicationLinks } = this.props;
+
+    if (!isCompact) {
+      if (applicationLinks && applicationLinks.length) {
+        return (
+          <ApplicationTabs links={applicationLinks} />
+        );
+      }
+
+      return null;
+    }
+
+    /**
+     * When compact, the navigation region of the header renders the application name component instead.
+     */
+    return this.renderAppName();
+  }
+
+  renderExtensions(isCompact) {
+    const { app, layoutConfig, extensions } = this.props;
+
+    if (!isCompact && extensions) {
+      return React.cloneElement(extensions, { app, layoutConfig });
+    }
+
+    return null;
+  }
+
+  renderUtilities(isCompact) {
+    const { utilityConfig } = this.props;
+
+    if (!isCompact && utilityConfig) {
+      return (
+        <ApplicationHeaderUtility
+          onChange={this.handleUtilityOnChange}
+          onDisclose={this.handleUtilityDiscloseRequest}
+          title={utilityConfig.title}
+          accessory={utilityConfig.accessory}
+          menuItems={utilityConfig.menuItems}
+          initialSelectedKey={utilityConfig.initialSelectedKey}
+          data-application-header-utility
+        />
+      );
+    }
+
+    return null;
+  }
+
+  renderUtilitiesPopup() {
+    const { utilityComponent } = this.state;
+
+    if (utilityComponent) {
+      return (
+        <Popup
+          attachmentBehavior="none"
+          contentAttachment="top center"
+          contentHeight="auto"
+          contentWidth="240"
+          isArrowDisplayed
+          isHeaderDisabled
+          isOpen
```

Comment:
Should this always be open?

Reply:
@dkasper-was-taken would it make more sense to render the popup at all times and flex its children/isOpen props based on state?

Reply:
bump